src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1277,14 +1282,22 @@ protected override void HandleStackTraceRequestAsync(IRequestResponder<StackTrac
                             textPosition = TextPositionTuple.GetTextPositionOfFrame(m_pathConverter, frame) ?? TextPositionTuple.Nil;
                         }
 
-                        response.StackFrames.Add(new ProtocolMessages.StackFrame()
+                        // remember the state for the lowermost frame
+                        if (startFrame == 0 && i == 0)
+                            m_currentFrameHasSourceCode = textPosition.Source?.Path != null;
```

Comment:
>m_currentFrameHasSourceCode [](start = 28, length = 27)

You will wind up setting this to true if the top frame of any thread doesn't have source. It needs to be changed to a Dictionary/HashSet. I would probably also suggest changing the name from 'current' to 'top' since 'current frame' has another meaning that can be confusing.

Reply:
Depends on the UI I guess. VSCode probably fetches only the current thread so it works.

Reply:
No, VS Code fetches the stack of all threads very aggressively.

Reply:
Ok. Haven't noticed any fallout yet though.
While without the if I had. All frames of a thread were marked.